Lib/fontTools/subset/__init__.py

```diff
@@ -2610,12 +2612,17 @@ def _subset_glyphs(self, font):
 				log.warning("%s NOT subset; don't know how to subset; dropped", tag)
 				del font[tag]
 
-		if not self.options.retain_gids:
-			with timer("subset GlyphOrder"):
-				glyphOrder = font.getGlyphOrder()
+		with timer("subset GlyphOrder"):
+			glyphOrder = font.getGlyphOrder()
+			if not self.options.retain_gids:
 				glyphOrder = [g for g in glyphOrder if g in self.glyphs_retained]
-				font.setGlyphOrder(glyphOrder)
-				font._buildReverseGlyphOrderDict()
+			else:
+				max_retained_glyph_id = max(font.getGlyphID(g) for g in self.glyphs_retained)
```

Comment:
you could reuse the `self.last_retained_order` here instead of recomputing `max_retained_glyph_id`

Reply:
Good catch, fixed.